sdk/schemaregistry/azure-schemaregistry-avroencoder/azure/schemaregistry/encoder/avroencoder/_schema_registry_avro_encoder.py

```diff
@@ -170,77 +175,87 @@ def encode(
                 f"Cannot parse schema: {raw_input_schema}", error=e
             ).raise_with_traceback()
 
-        schema_id = self._get_schema_id(schema_fullname, raw_input_schema)
+        cache_misses = self._get_schema_id.cache_info().misses  # pylint: disable=no-value-for-parameter
+        request_kwargs = request_kwargs or {}
+        schema_id = self._get_schema_id(schema_fullname, raw_input_schema, **request_kwargs)
+        new_cache_misses = self._get_schema_id.cache_info().misses  # pylint: disable=no-value-for-parameter
+        if new_cache_misses > cache_misses:
+            cache_size = self._get_schema_id.cache_info().currsize  # pylint: disable=no-value-for-parameter
+            _LOGGER.info("New entry has been added to schema ID cache. Cache size: %s", str(cache_size))
```

Comment:
I'm curious about how this works: 

as per my understanding, `cache_info().misses` is a monotonic increasing value -- whenever we miss a key, misses += 1, and we only log is there's a miss.

with that being said, in a single call new_cache_misses  could only be
- cache_misses + 1 if there's a miss
- cache_misses if there's no miss

am my understanding right?

Reply:
yes that's correct. Miss + 1 anytime a new entry has been added to the cache (i.e. there has not been a collision or "hit" by an entry to the cache).